test/auth-source-pass-tests.el

```diff
@@ -310,6 +310,18 @@ HOSTNAME, USER and PORT are passed unchanged to
       (auth-source-pass--find-match "bar.com" nil nil)
       (should (= (length auth-source-pass--parse-log) 1)))))
 
+(ert-deftest auth-source-pass--find-match-most-specific-suffix-is-invalid ()
+  (auth-source-pass--with-store
+   '(("foo@bar.com") ("bar.com" ("key" . "val")))
+   (should (consp (auth-source-pass--find-match "bar.com" "foo" nil)))
+   (should (equal auth-source-pass--parse-log '("bar.com" "foo@bar.com")))))
```

Comment:
I think the test would be clearer if the 2 entries where in the same order as they are parsed. This requires changing the macro though.